packages/core/src/StateNode.ts

```diff
@@ -222,6 +196,14 @@ export class StateNode<
     this.tags = toArray(config.tags);
   }
 
+  public initialize() {
```

Comment:
this should be private and if it can't be easily made private cause of typing reasons then we should at least prefix this with an underscore

A separate note is that it would be great if we could remove this one in the future. The "resolved" shape of a transition is only needed first when we try to select it - no reason to resolve **all** of the transitions upfront. For huge machines the cost here can add up and it would be great if we could minimize it.